app/src/main/java/com/asfoundation/wallet/ui/iab/EarnAppcoinsPresenter.kt

```diff
@@ -0,0 +1,32 @@
+package com.asfoundation.wallet.ui.iab
+
+import io.reactivex.Observable
+import io.reactivex.Scheduler
+import io.reactivex.disposables.CompositeDisposable
+
+class EarnAppcoinsPresenter(private val view: EarnAppcoinsView,
+                            private val disposables: CompositeDisposable,
+                            private val viewScheduler: Scheduler) {
+  fun present() {
+    handleBackClick()
+    handleDiscoverClick()
+  }
+
+  private fun handleDiscoverClick() {
+    disposables.add(view.discoverButtonClick()
+        .observeOn(viewScheduler)
+        .doOnNext { view.navigateToAptoide() }
+        .subscribe({}, { it.printStackTrace() }))
```

Comment:
**Discussion** 
Since doOnNext has an different instruction shouldn't we use the same for doOnError?

        .doOnNext { view.navigateBack(PaymentMethodsView.SelectedPaymentMethod.EARN_APPC) }
        .doOnError { it.printStackTrace() }
        .subscribe())